api/controllers/ExampleController.js

```diff
@@ -31,10 +31,10 @@ module.exports = {
       topic_id: topicId
     }).then(function(instance) {
       RedisClient.del('view_topic_' + topicId);
-      if(req.wantsJSON) {
+      if(req.wantsJSON && !(req.param('viewer_pane')==1)) {
```

Comment:
I think it would be cleaner to request html from RStudio instead of adding a condition here.
You can do that by setting the `Accept` header to `text/html` in the ajax request